src/main/kotlin/org/rust/ide/intentions/ConvertMethodCallToUFCSIntention.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.ide.inspections.import.AutoImportFix
+import org.rust.ide.inspections.import.import
+import org.rust.ide.presentation.renderInsertionSafe
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.types.ty.Mutability
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyReference
+import org.rust.lang.core.types.type
+
+class ConvertMethodCallToUFCSIntention : RsElementBaseIntentionAction<ConvertMethodCallToUFCSIntention.Context>() {
+    override fun getText() = "Convert to UFCS"
+    override fun getFamilyName() = text
+
+    override fun findApplicableContext(
+        project: Project,
+        editor: Editor,
+        element: PsiElement
+    ): Context? {
+        val methodCall = element.parentOfType<RsMethodCall>(true) ?: return null
```

Comment:
Let's also limit the intention scope to a method identifier. I.e. make it available only when the caret is placed next to a method identifier. Otherwise, it can be too annoying in a large method calls like
```rust
foo.bar(
    baz.quux(spam.eggs(0)),
    foobar.quuux(a + b.get_value())
)
```